src/pytest_workflow/plugin.py

```diff
@@ -129,13 +142,78 @@ def pytest_collection(session):
     print()
 
 
-def pytest_runtestloop(session):
+def pytest_collection_modifyitems(config: PytestConfig,
+                                  items: List[pytest.Item]):
+    """Here we skip all tests related to workflows that are not executed"""
+
+    for item in items:
+        marker = item.get_closest_marker(
+            name="workflow")  # type: Optional[MarkDecorator] # noqa: E501
+        if marker is not None:
+            workflow_name = marker.kwargs.get('name')
+            if workflow_name is None:
+                # If name key is not defined use the first arg.
+                if len(marker.args) >= 1:
+                    workflow_name = marker.args[0]
+                    # Make sure a name attribute is added anyway for the
+                    # fixture lookup.
+                    marker.kwargs['name'] = workflow_name
+                else:
+                    # If we raise an error here a number of things will happen:
+                    # + Pytest will crash. Giving a lot of INTERNAL ERROR lines
+                    # + No tests will run
+                    # + No workflows will be started
+                    # + All the threads that are waiting for a workflow to
+                    #   finish will wait indefinitely. Causing an infinite
+                    #   hang. Pytest will never finish.
+                    # Therefore we do not crash here, but raise a warning.
+                    item.warn(pytest.PytestWarning(
+                        "A workflow name should be defined in the "
+                        "workflow marker of {0}".format(item.nodeid)))
+                    # Go on with the next item.
+                    continue
+
+            if workflow_name not in config.executed_workflows:
+                skip_marker = pytest.mark.skip(
+                    reason="'{0}' has not run.".format(workflow_name))
+                item.add_marker(skip_marker)
+
+
+def pytest_runtestloop(session: pytest.Session):
     """This runs after collection, but before the tests."""
     session.config.workflow_queue.process(
         session.config.getoption("workflow_threads")
     )
 
 
+def pytest_sessionfinish(session: pytest.Session):
+    if not session.config.getoption("keep_workflow_wd"):
+        for tempdir in session.config.workflow_cleanup_dirs:
+            shutil.rmtree(str(tempdir))
+
+
+@pytest.fixture()
+def workflow_dir(request: SubRequest):
+    """Returns the workflow_dir of the workflow named in the mark. This fixture
+    is only provided for tests that are marked with the workflow mark."""
+
+    # request.node refers to the node that has the mark. This is a pytest.Node
+    marker = request.node.get_closest_marker(name="workflow")
+
+    if marker is not None:
+        workflow_temp_dir = request.config.workflow_temp_dir
+        try:
+            workflow_name = marker.kwargs['name']
+        except KeyError:
+            raise ValueError(
```

Comment:
This should be `TypeError`, as that is python's default exception for when there is call with too few arguments.

Reply:
:+1: